examples/counter/src/contract.rs

```diff
@@ -0,0 +1,294 @@
+
+// Using the Rust macro cw_orch::interface provided by cw-orchestrator we can define our contract entry points.
+// This also generates a struct using our contract cargo name using PascalCase.
+// In this example the name is CounterContract.
+// This macro helps us with basic logic, keeps our contracts DRY and more important, it helps us speed our development process up
+#[cw_orch::interface]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: msgs::InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    COUNT.save(deps.storage, &Count(msg.initial_value))?;
+
+    Ok(Response::default().add_attribute("initial_value", msg.initial_value.to_string()))
+}
+
+#[cw_orch::interface]
+pub fn query(deps: Deps, _env: Env, msg: msgs::QueryMsg) -> StdResult<Binary> {
+    match msg {
+        msgs::QueryMsg::GetCount => Ok(to_binary(&CurrentCount(COUNT.load(deps.storage)?.0))?),
+    }
+}
+
+#[cw_orch::interface]
+pub fn execute(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: msgs::ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let response = match msg {
+        msgs::ExecuteMsg::Increase => {
+            let mut value = COUNT.load(deps.storage)?.0;
+            value = value.checked_add(1u128.into())?;
+            COUNT.save(deps.storage, &Count(value))?;
+            Response::default().add_attribute("action", "increase")
+        }
+        msgs::ExecuteMsg::Decrase => {
+            let mut value = COUNT.load(deps.storage)?.0;
+            value = value.checked_sub(1u128.into())?;
+            COUNT.save(deps.storage, &Count(value))?;
+            Response::default().add_attribute("action", "decrease")
+        }
+        msgs::ExecuteMsg::IncreaseBy(amount) => {
+            let mut value = COUNT.load(deps.storage)?.0;
+            value = value.checked_add(amount.into())?;
+            COUNT.save(deps.storage, &Count(value))?;
+            Response::default().add_attribute("action", "increase_by")
+        }
+    };
+
+    Ok(response)
+}
+
+#[cw_orch::interface]
+pub fn migrate(
+    deps: DepsMut,
+    _env: Env,
+    msg: msgs::MigrateMsg<msgs::InstantiateMsg>,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, msg.version)?;
+    COUNT.save(deps.storage, &Count(msg.conf.initial_value))?;
+    Ok(Response::default().add_attribute("action", "migrate"))
+}
+
+// Now that we have setup for our contract entry points, We can continue to the next step.
+// This is where more of the magic of cw-orchestrator occurs
+// In this case we will prepare a trait for our two scenarios Mock and Daemon
+// Daemon is our production scenario, deploying to a real blockchain, be it a local testnet, a tesnet our a mainnet
+// and Mock is our development scenario, used for unit testing and fine tuning our contract with speed
+trait CounterWrapper<T: TxHandler> {
```

Comment:
Can you explain why you create a trait for this and why it has this structure? 
The `interface` macros create a struct that is essentially already a wrapper. 

How will this help devs to understand cw-orch?

Reply:
hmm I honestly dont think this will help devs understand cw-orch more.
Added it to keep the example DRY.
What would you rather have here?, just repeat code in the implementations below?

I also think that maybe this pattern is going to be a common approach for many devs.
Because they dont know that or even care?, and will try to keep their code DRY too or just follow a similar approach...
IDK